### DISCUSSION TITLE
Update number to number +1

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#2230] [Q&A] [answered]
Hi,

Sorry for asking such a silly question but i have user with the info:

`{
  avatar: 'aniyuki_sad_anime_avatar_image_49_3tNIr5smWj.jpg',
  collectionId: '_pb_users_auth_',
  collectionName: 'users',
  created: '2023-03-22 01:15:20.180Z',
  email: 'test@hotmail.de',
  emailVisibility: false,
  id: 'db5s41oatgoeh0q',
  karma: 0,
  name: 'test',
  updated: '2023-04-04 15:03:01.603Z',
  username: 'test',
  verified: false,
  expand: {}
}`

I want to find the user and then increase the karma point to +1. 

`let getuser = await locals.pb
				.collection("users")
				.getOne(locals.user.id)
				.update(locals?.user?.karma, locals?.user?.karma+1);`

How can i do it in a simple way

**Comment by ganigeorgiev** [comment, marked as answer]
If you already know the id of the record, you don't necessarily need a `getOne` request. Just `update` with the data you want to submit as second argument should be fine. If you want to perform "increment" type of operation without explicitly specifying the previous value, you can append the `+` modifier to the field name, something like:

```js
await local.pb.collection("users").update(locals.user.id, {
    "karma+": 1,
})
```

**Reply by MuslemRahimi** [reply]
You're the best

**Comment by txstc55** [comment]
Just a question: if I only allow field to be incremented, can I write in api rules:
```javascript
@request.data.karma:isset = false
```
Would that disallow me incrementing karma or does `karma+` and `karma` are treated as two different fields?

**Reply by ganigeorgiev** [reply]
`karma+` is not a new field and it is treated the same as `karma`, otherwise it will require users to explicitly list every possible modifier in their rule and this will be too verbose and prone to error. The only difference is that `karma+` is computed/resolved server-side before actual processing.

There is no rule currently that will allow only add/subtract and disallow direct value set and I'm not sure if that will be useful in general, because whether the user will submit `karma=30` or `karma+:25` from security perspective is not much different, the final value that is being persisted should be the important bit.
